### PR TITLE
[finished #171928157] Adding initial glossary of terms

### PR DESCRIPTION
**Pull request by simonwa7**

### CODE REVIEW

File: intro/content/basics/glossary.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+title: Glossary
+description: A list of helpful definitions for Orquestra
+---
+
+- `Workflow`: An instruction pattern of steps to be performed that is easily reproduceable, distributed, and shareable. 
+- `workflowID`: A randomly generated string that is used as an identifier for a specific run of a workflow.
```
Comment (jama22) [inline intro/content/basics/glossary.md:7]: worth mentioning if `workflowID` is unique or not unique

File: intro/content/basics/glossary.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+- `Source code`: Code contained in a resource that is callable by resource templates or source code contained in other resources.
+- `Resource Template`: A template that is contained in a resource and, when called, executes a script that calls source code. 
+- `Workflow Template`: A template that is typically contained inside a workflow that, when called, executes a series of steps.
+- `Artifact`: A data object - that is processed and can be shared between different steps in a workflow - that is serialized and represented in a file that can be written to disk (e.g. JSON). 
+- `Workflow result`: A file produced by the `Data Correlation Service` that contains data representing all steps and data representing each artifact produced by each step in a workflow. It can easily be used to analyze the data produced by your workflow.
+- `Step`: An execution process in a workflow - performed by invoking a template.
+- `StepID`: A randomly generated string that is used as an identifier for a specific step.
```
Comment (jama22) [inline intro/content/basics/glossary.md:17]: same as `workflowID`, worth mentioning if its unique / not-unique

File: intro/content/basics/glossary.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+---
+title: Glossary
+description: A list of helpful definitions for Orquestra
+---
+
```
Comment (jama22) [inline intro/content/basics/glossary.md:5]: Is it worth creating sub-categories in this Glossary to group concepts? Some terms are specific to the act of writing `workflow.yaml` and others are about the running of workflows in QE. Just a thought, the list is short enough that we can get away with just a long list for now

Comment (simonwa7) [reply intro/content/basics/glossary.md:5]: Yeah my thought was the list was short enough to leave as one long one for now. I was thinking if it gets longer maybe we'd have glossaries for the different components of orquestra (ie one on Quantum Engine, one on Data Management, etc)

File: intro/content/basics/glossary.md (hunk 2 above)
Comment (simonwa7) [reply intro/content/basics/glossary.md]: Addressed in commit below

File: intro/content/basics/glossary.md (hunk 1 above)
Comment (simonwa7) [reply intro/content/basics/glossary.md]: Addressed in commit below

**Comment by jama22** [comment]
lgtm

**Pull request merged**